pkg/types/machinepools.go

```diff
@@ -9,6 +9,8 @@ import (
 // MachinePool is a pool of machines to be installed.
 type MachinePool struct {
 	// Name is the name of the machine pool.
+	// For the control plane machine pool, the name will always be "master".
+	// For the compute machine pools, the only valid name is "worker".
```

Comment:
It would be nice to lift this restriction.  All we're doing for compute nodes is pushing MachineSets which reference an installer-created worker IAM instance profile, right?  We can do that regardless of the name we set in the MachineSet.

Reply:
Yes, this restriction will be lifted. This is just the first step in a larger transition that will include that lifted restriction.